lib/msal-angular/src/msal.interceptor.ts

```diff
@@ -257,51 +263,74 @@ export class MsalInterceptor implements HttpInterceptor {
   private matchResourcesToEndpoint(
     protectedResourcesEndpoints: string[],
     endpoint: string
-  ): Array<string> {
-    const matchingResources: Array<string> = [];
+  ): MatchingResources {
+    const matchingResources: MatchingResources = {
+      absoluteResources: [],
+      relativeResources: [],
+    };
+
+    // Used to store the resource with the index position in the endpoint URL
+    const absoluteResourcesWithIndex = new Map<string, number>();
+    // Ensures endpoints and protected resources compared are normalized
+    const normalizedEndpoint = this.location.normalize(endpoint);
 
     protectedResourcesEndpoints.forEach((key) => {
+      // Normalizes and adds resource to matchingResources.absoluteResources if key matches endpoint. StringUtils.matchPattern accounts for wildcards
       const normalizedKey = this.location.normalize(key);
+      const absoluteKey = this.getAbsoluteUrl(key);
 
-      // Get url components
-      const absoluteKey = this.getAbsoluteUrl(normalizedKey);
-      const keyComponents = new URL(absoluteKey);
-      const absoluteEndpoint = this.getAbsoluteUrl(endpoint);
-      const endpointComponents = new URL(absoluteEndpoint);
+      // Storing the resource with the index position in the endpoint URL
+      if (StringUtils.matchPattern(normalizedKey, normalizedEndpoint)) {
+        const indexn = endpoint.indexOf(normalizedKey);
+        absoluteResourcesWithIndex.set(key, Math.max(0, indexn));
```

Comment:
Why `Math.max` here instead of directly using `indexn`? What's the scenario where `indexn` is < 0?

Reply:
I also don't quite understand what's happening here w.r.t. index. Why are you looking for the index of the key, then later sorting by index? I see comments around _what_ is happening but it would be helpful to also include comments about _why_

Reply:
In issue #7111 , the user had reported a sample request like 
_https://MY_API_SITE_1/api/sites?$filter=siteUrl eq 'https://MY_API_SITE_2'_
which was failing, so this logic matches the pattern for key in the URL and tries to pick the first match, I am just storing the index as 0. All this logic to handle the request mentioned in  #7111. It was difficult to explain in comments as these requests with kind of sub URL is rare. Hope you could help me with the comments. 
You may check the test cases added by jo-arroyo for #7111 fix. this logic targets that fix in a way such that it does not break existing matching logic with path-scopes matching.